httpclient/request.go

```diff
@@ -25,35 +25,40 @@ import (
 // Request bodies are never closed by the client, hence only accepting
 // io.Reader.
 type requestBody struct {
-	Reader     io.Reader
-	DebugBytes []byte
+	Reader      io.Reader
+	ContentType string
+	DebugBytes  []byte
 }
 
 func newRequestBody(data any) (requestBody, error) {
 	switch v := data.(type) {
 	case io.Reader:
 		return requestBody{
-			Reader:     v,
-			DebugBytes: []byte("<io.Reader>"),
+			Reader:      v,
+			ContentType: "application/octet-stream",
```

Comment:
Let's not set ContentType for anything else then. Agreed on JSON, but we can default to no header for other request bodies.